### ISSUE TITLE
PMcycle and PMfinal objects not saved

### ISSUE THREAD

**Issue by mhovd** [LAPKB/Pmetrics#58]
Following an upgrade to R 4.2.0, and a fresh install of Pmetrics 1.9.7, I have not been able to save the cycle and final objects following a run. This error also occurs when using the built-in model (`Pmetrics::model`) and dataset (`Pmetrics::mdata.1`).

Following I run, I get the following errors in the terminal:
```
WARNING: error in extraction of cycle information; 'PMcycle' object not saved.
WARNING: error in extraction of final cycle parameter values; 'PMfinal' object not saved.
```

If I try to make the objects manually, I get the following errors:
```r
npag = NPparse(outfile = "NP_RF0001.TXT")
> makeCycle(npag)
Error in if (class(data$isd) == "numeric") data$isd <- matrix(data$isd,  : 
  the condition has length > 1
> makeFinal(npag)
Error in if (!is.na(popCor)) { : the condition has length > 1
```

R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.3.1

**Comment by Siel** [comment]
I've been trying but I cannot reproduce this error. It is only happening on Mac with Intel chips?

**Comment by mhovd** [comment]
I have not been able to test this on any other systems yet, but the error is still present on my MacBook with Intel chip.

**Comment by christyny** [comment]
I'm running under Windows 10, but the problem still occurs.

**Comment by mhovd** [comment]
The issue is still present after re-installation of Pmetrics (latest commit `40f12edbf2df109bc57b1d6b7f42438250674464 `, R 4.2.0, Monterey 12.4).
Additionally, the `cov.*` object is not saved under some occasions, resulting in the following errors:

```
WARNING: error in extraction of cycle information; 'PMcycle' object not saved.
WARNING: error in extraction of final cycle parameter values; 'PMfinal' object not saved.
WARNING: error in extraction of covariate-parameter data; 'PMcov' object not saved.
```

The additional error was apparent when including a single patient only, and using a prior. I have not yet been able to isolate the reason for the missing `cov.*`-object, or been able to make a reproducible example outside of this particular project. However, the same project was able to save all files previously, suggesting the errors might be related.

**Comment by Siel** [comment]
@mhovd I'm still unable to reproduce the error, could you please check if that bug is present int the 'Version2' branch?

**Comment by mhovd** [comment]
> @mhovd I'm still unable to reproduce the error, could you please check if that bug is present int the 'Version2' branch?

Unfortunately the error is still present in the `Version2` branch, here is the error message from Terminal
```
WARNING: error in extraction of cycle information; 'PMcycle' object not saved.
WARNING: error in extraction of final cycle parameter values; 'PMfinal' object not saved.
WARNING: error in extraction of covariate-parameter data; 'PMcov' object not saved.

[1] "../inputs/mrgmodel.txt"

Saving R data objects to /Users/markushovd/Prosjekter/Iohexol/Runs mrgdata ind/1/inputs/1/outputs......

Use PMload() to load them.

The following objects have been saved:

NPdata: All output from NPAG
pop: Population predictions at regular, frequent intervals
post: Posterior predictions at regular, frequent inteverals
op: Observed vs. population and posterior predicted
mdata: The data file used for the run
Error in if (omit == 0) omit <- 1 : argument is of length zero
Calls: PMreport ... toHTML -> resolveSizing -> <Anonymous> -> plot.PMcycle
In addition: Warning message:
In stringr::str_split(err[i + 1], ",")[[1]] %>% as.numeric() :
  NAs introduced by coercion
Execution halted
The file example/1/inputs/1/outputs/NPAGreport.html does not exist.
```

And the same error as previous when attempting to create the objects manually.
```
> makeFinal(np)
Error in if (!is.na(popCor)) { : the condition has length > 1
```

```
sessionInfo()
R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.9     purrr_0.3.4    
 [5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.7    ggplot2_3.3.6  
 [9] tidyverse_1.3.1 Pmetrics_2.0.0 

loaded via a namespace (and not attached):
 [1] mclust_5.4.10     lubridate_1.8.0   prettyunits_1.1.1 ps_1.7.0         
 [5] rprojroot_2.0.3   assertthat_0.2.1  foreach_1.5.2     utf8_1.2.2       
 [9] R6_2.5.1          cellranger_1.1.0  backports_1.4.1   reprex_2.0.1     
[13] httr_1.4.3        pillar_1.7.0      rlang_1.0.2       curl_4.3.2       
[17] readxl_1.4.0      rstudioapi_0.13   callr_3.7.0       desc_1.4.1       
[21] devtools_2.4.3    munsell_0.5.0     broom_0.8.0       compiler_4.2.0   
[25] modelr_0.1.8      pkgconfig_2.0.3   base64enc_0.1-3   pkgbuild_1.3.1   
[29] tidyselect_1.1.2  gridExtra_2.3     codetools_0.2-18  fansi_1.0.3      
[33] crayon_1.5.1      tzdb_0.3.0        dbplyr_2.1.1      withr_2.5.0      
[37] brio_1.1.3        grid_4.2.0        jsonlite_1.8.0    gtable_0.3.0     
[41] lifecycle_1.0.1   DBI_1.1.2         magrittr_2.0.3    scales_1.2.0     
[45] cli_3.3.0         stringi_1.7.6     cachem_1.0.6      cubelyr_1.0.1    
[49] remotes_2.4.2     fs_1.5.2          testthat_3.1.4    xml2_1.3.3       
[53] ellipsis_0.3.2    generics_0.1.2    vctrs_0.4.1       iterators_1.0.14 
[57] tools_4.2.0       glue_1.6.2        hms_1.1.1         pkgload_1.2.4    
[61] processx_3.5.3    fastmap_1.1.0     colorspace_2.0-3  sessioninfo_1.2.2
[65] rvest_1.0.2       memoise_2.0.1     haven_2.5.0       usethis_2.1.5    
```

**Comment by Siel** [comment]
@mhovd I'm using R 4.2 and still i'm unable to reproduce the error, but this user reinstalled R 4.1.3 and got it fixed: https://github.com/LAPKB/Pmetrics/discussions/64#discussioncomment-2833080. Could you try that?

**Comment by mhovd** [comment]
> @mhovd I'm using R 4.2 and still i'm unable to reproduce the error, but this user reinstalled R 4.1.3 and got it fixed: [#64 (reply in thread)](https://github.com/LAPKB/Pmetrics/discussions/64#discussioncomment-2833080). Could you try that?

I can try that, but I would like to use R 4.2 for a couple of packages that I need.
However, I was able to reproduce the error on my machine using the latest image of `rocker/rstudio`
```
# Run in terminal
docker run -e PASSWORD=testing -p 8787:8787 rocker/tidyverse

# Open RStudio in browser at localhost:8787, username=rstudio, password=testing

# Install Pmetrics (main branch) as usual with devtools
writeLines(model, "model.txt")
PMwriteMatrix(mdata.1, "data.csv")

NPrun(intern = TRUE)
```

**Comment by Siel** [comment]
@mhovd Actually we are going to release a docker version of pmetrics, see: https://hub.docker.com/r/lapkb/pmetrics/tags

![image](https://user-images.githubusercontent.com/1023006/170828235-01f73b55-a139-4f9b-866f-d4682be11476.png)
![image](https://user-images.githubusercontent.com/1023006/170828260-754b994f-e2d3-4a5e-be3d-0c4a23cf4896.png)
![image](https://user-images.githubusercontent.com/1023006/170828270-36c85ac8-4c2a-4aef-b5ea-553b8cc08952.png)

As you see, the same example and it is working for me, I tried using Pmetrics 1.9.7 and Pmetrics 2.0, both work fine

**Comment by mhovd** [comment]
I suspect this error could be due to some change in R 4.2, as the Pmetrics Docker image seems to use R 4.1.2
Were you able to try with the `rocker/rstudio` image as well? 

EDIT: Removed unnecessary information.

**Comment by mhovd** [comment]
I was able to test this on my desktop (Windows 10, Intel) as well, with a fresh install of R 4.2 and Pmetrics  1.9.7 and 2.0.0 - the same error was present on both.

**Comment by mhovd** [comment]
> @mhovd I'm using R 4.2 and still i'm unable to reproduce the error, but this user reinstalled R 4.1.3 and got it fixed: [#64 (reply in thread)](https://github.com/LAPKB/Pmetrics/discussions/64#discussioncomment-2833080). Could you try that?

I temporarily installed R 4.1.3 and that seemed to solve the problem.

**Comment by Siel** [comment]
Thanks @mhovd, we might need to address this soon, we have plans to release Pmetrics 2.0 soon, I'll try to fix this problem with R 4.2 by then

**Comment by Siel** [comment]
@mhovd, I just fixed the issue in the Version2 branch, could you check? I'm going to close the issue for now, but feel free to reopen it if you re-experience the issue

**Comment by mhovd** [comment]
@Siel, could this fix be applied to the legacy version in the `main` branch as well?

**Comment by Siel** [comment]
hi @mhovd, yes. Actually we just released 2.0. So the fix is in the main branch.
